docs/format/db.rst

```diff
@@ -67,3 +103,35 @@ Example line: bit 01_256
 
 See previous section for number meaning
 
+For each segment type there is a mask file `mask_<seg_type>.db` that contains one line for each bit that has been observed being set in any of the example designs generated during generation of the database. The lines simply contain the keyword `bit` followed by the bit position. This database is used to identify unused bits in the configuration segments.
+
+
+.bits example
+-------------
+
+Say entry is: bit_0002050b_002_05
+
+2 step process:
+* Decode which segment
+* Decode which bit within that segment
+
+We have:
+* Frame address 0x0002050b (hex)
+* Word #: 2 (decimal, 0-99)
+* Bit #: 5 (decimal, 0-31)
+
+The CLB tile and the associated interconnect switchbox tile are configured together as a segment. However, configuration data is grouped by segment column rather than tile column. First, note this segment consists of 36 frames. Second, note there are 100 32 bit words per frame (+ 1 for checksum => 101 actual). Each segment takes 2 of those words meaning 50 segments (ie 50 CLB tiles + 50 interconnect tiles) are effected per frame. This means that the smallest unit that can be fully configured is a group of 50 CLB tile + switchbox tile segments taking 4 * 36 * 101 = 14544 bytes. Finally, note segment columns are aligned to 0x80 addresses (which easily fits the 36 required frames).
+
+tilegrid.json defines addresses more precisely. Taking 0x0002050b, the frame base address is 0x0002050b & 0xFFFFFF80 => 0x00020500. The frame offset is 0x0002050b & 0x7F => 0x0B => 11.
+
+So in summary:
+* Frame base address: 0x00020500
+* Frame offset: 0x0B (11)
+* Frame word #: 2
+* Frame word bit #: 5
+
+So, with this in mind, we have frame base address 0x00020500 and word # 2. This maps to tilegrid.json entry SEG_CLBLL_L_X12Y101 (has "baseaddr": ["0x00020600", 2]). This also yields "type": "clbll_l" meaning we are configuring a CLBLL_L.
```

Comment:
This example is out of date with the new tilegrid format, should update it.

Reply:
Added a FIXME.